Sources/Controller/Hits/CollectionView/HitsCollectionViewDataSource.swift

```diff
@@ -11,21 +11,36 @@ import UIKit
 open class HitsCollectionViewDataSource<DataSource: HitsSource>: NSObject, UICollectionViewDataSource {
   
   public var cellConfigurator: CollectionViewCellConfigurator<DataSource.Record>
+  public var templateCellProvider: () -> UICollectionViewCell
   public weak var hitsSource: DataSource?
   
   public init(cellConfigurator: @escaping CollectionViewCellConfigurator<DataSource.Record>) {
     self.cellConfigurator = cellConfigurator
+    self.templateCellProvider = { return .init() }
   }
   
   open func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-    return hitsSource?.numberOfHits() ?? 0
+    
+    guard let hitsSource = hitsSource else {
+      fatalError("Missing hits source")
```

Comment:
One way we can improve the wording on this is: instead of just stating what is the problem, we can propose to them the solution to give them context. 

Example: 
> The HitSource datasource has not been set. You need to set it by setting the `hitsCollectionViewDataSource.hitsSource` property.